csharp/src/Drivers/Apache/Hive2/HiveServer2Reader.cs

```diff
@@ -78,6 +107,55 @@ static IArrowArray GetArray(TColumn column)
                 (IArrowArray?)column.StringVal?.Values ??
                 (IArrowArray?)column.BinaryVal?.Values ??
                 throw new InvalidOperationException("unsupported data type");
+            if (expectedArrowType != null && arrowArray is StringArray stringArray && s_arrowStringConverters.ContainsKey(expectedArrowType.TypeId))
+            {
+                // Perform a conversion from string to native/scalar type.
+                Func<StringArray, IArrowType, IArrowArray> converter = s_arrowStringConverters[expectedArrowType.TypeId];
+                return converter(stringArray, expectedArrowType);
+            }
+            return arrowArray;
+        }
+
+        private static Date32Array ConvertToDate32(StringArray array, IArrowType _)
+        {
+            var resultArray = new Date32Array.Builder();
+            foreach (string item in (IReadOnlyCollection<string>)array)
+            {
+                resultArray.Append(DateTime.Parse(item));
+            }
+
+            return resultArray.Build();
+        }
+
+        private static Decimal128Array ConvertToDecimal128(StringArray array, IArrowType schemaType)
+        {
+            // Using the schema type to get the precision and scale.
+            var resultArray = new Decimal128Array.Builder((Decimal128Type)schemaType);
+            foreach (string item in (IReadOnlyList<string>)array)
+            {
+                // Trying to parse the value into a decimal to handle the exponent syntax. But this might overflow.
+                if (decimal.TryParse(item, NumberStyles.Float, CultureInfo.InvariantCulture, out decimal decimalValue))
+                {
+                    resultArray.Append(new SqlDecimal(decimalValue));
+                }
+                else
+                {
+                    resultArray.Append(item);
```

Comment:
This overload uses the "built-in" parsing in `Decimal128Array` - but it doesn't handle exponent notation.

Reply:
Consider adding as a comment, though if there's a test looking for 'E' or 'e' I think that might be enough of a signal to explain why.

Reply:
I've replace this with a call to my implementation of the `GetBytes` method that also handles exponent notation. It is as performant as the `Apache.Arrow.DecimalUtility.GetBytes(string ...)` method.